azkaban-common/src/main/java/azkaban/imagemgmt/daos/HPFlowDaoImpl.java

```diff
@@ -0,0 +1,169 @@
+package azkaban.imagemgmt.daos;
+
+import azkaban.db.DatabaseOperator;
+import azkaban.db.SQLTransaction;
+import azkaban.executor.ExecutableFlow;
+import azkaban.imagemgmt.exception.ErrorCode;
+import azkaban.imagemgmt.exception.ImageMgmtDaoException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.apache.log4j.Logger;
+
+import static azkaban.imagemgmt.utils.ErroCodeConstants.*;
+
+
+/**
+ * Dao implementation for managing high priority flows
+ */
+@Singleton
+public class HPFlowDaoImpl implements HPFlowDao {
+  private static final Logger log = Logger.getLogger(HPFlowDaoImpl.class);
+
+  private final DatabaseOperator databaseOperator;
+
+  // Flow ID is a fully qualified flow name with format,
+  // <project_name>.<flow_ame>
+  private static String INSERT_HP_FLOW_OWNER =
+      "INSERT into hp_flow_owners (name, created_on, created_by, "
+      + "modified_on, modified_by ) "
+      + "values (?, ?, ?, ?, ?)";
+
+  private static String INSERT_HP_FLOWS =
+      "INSERT into hp_flows (flow_id, created_on, created_by) "
+      + "values (?, ?, ?)";
+
+  @Inject
+  public HPFlowDaoImpl(final DatabaseOperator databaseOperator) {
+    this.databaseOperator = databaseOperator;
+  }
+
+  /**
+   * Returns if the provided user has high priority flow management access.
+   * @param userId : owner of high priority flows.
+   * @return
+   */
+  @Override
+  public boolean isHPFlowOwner(final String userId) {
+    final FetchHPFlowOwnership fetchHPFlowOwnership = new FetchHPFlowOwnership();
```

Comment:
Is it necessary to create a new object every time?

Reply:
I am not sure, I believe I followed the pattern used in the rest of the code. Do you see any harm in doing this?

Reply:
My understanding is that in general dependencies should be injected. I don't see a specific issue here, but I think it might be better memory management to only create the object as necessary, especially since it seems like we're using this in a mostly static context (only to access the encapsulated query).